force-app/main/default/classes/Payment.cls

```diff
@@ -0,0 +1,64 @@
+public abstract class Payment implements IPayment {
+
+
+         //Static Variables to hold the various messages we use
+    public static final String SUCCESS_MESSAGE = 'Success!';
+    public static final String INVALID_AMOUNT = 'Amount should be greater than 0';
+    public static final String INVALID_ACCOUNT_NUMBER = 'Account number is invalid';
+    public static final String MISSING_PAYMENT_INFO = 'Payment details are missing';
+
+
+    //Use Static initilization to set our Valid Account Number Length
+    protected final Integer validAccountNumberLength{
+    get{ return validAccountNumberLength; }
+    set { validAccountNumberLength=value;}
+           
+    }
```

Comment:
No need to for the get set methods here. This will get set down stream a child class, so it'll get inherited when its extended.

Reply:
done